singer/utils.py

```diff
@@ -185,6 +185,9 @@ def check_config(config, required_keys):
     if missing_keys:
         raise Exception("Config is missing required keys: {}".format(missing_keys))
 
+def update_config_file(config_path, new_config):
+    with open(config_path, 'w') as output:
+        output.write(json.dumps(new_config, indent=2))
```

Comment:
I'm a bit concerned about the API here, due to the possibility of accidentally zeroing out the config.

I wonder if it would make more sense to take a key and a value, then add or update that in the file, instead of replacing the file itself? That way there is no way to delete things from the config, just add them as it needs.

Reply:
I agree. I like the idea of mirroring the `metadata` API here. https://github.com/singer-io/singer-python/blob/2cdb93f29b60a3bbb9e570248c48b39d66d18b9e/singer/metadata.py#L10-L23

Reply:
While we're talking about API design, one possible extension to the metadata approach, would be to have a 3-arity `write` that assumes that `breadcrumb = tuple()`. Could be nicer for simple uses of the API, at the cost of a bit more complexity in the code (type checking the second arg, etc).

I'm not familiar with the Pythonic way to do multi-arity, maybe just defining the function twice is enough, or it might have to be `*args` with explicit run-time checking code.

Reply:
I think pythonic here would be a default value for breadcrumb `breadcrumb=[]`.